freqtrade/templates/FreqaiExampleStrategy.py

```diff
@@ -0,0 +1,342 @@
+                informative[f"{coin}bb_lowerband-period_{t}"] = bollinger["lower"]
+                informative[f"{coin}bb_middleband-period_{t}"] = bollinger["mid"]
+                informative[f"{coin}bb_upperband-period_{t}"] = bollinger["upper"]
+
+                informative[f"%-{coin}bb_width-period_{t}"] = (
+                    informative[f"{coin}bb_upperband-period_{t}"]
+                    - informative[f"{coin}bb_lowerband-period_{t}"]
+                ) / informative[f"{coin}bb_middleband-period_{t}"]
+                informative[f"%-{coin}close-bb_lower-period_{t}"] = (
+                    informative["close"] / informative[f"{coin}bb_lowerband-period_{t}"]
+                )
+
+                informative[f"%-{coin}roc-period_{t}"] = ta.ROC(informative, timeperiod=t)
+                macd = ta.MACD(informative, timeperiod=t)
+                informative[f"%-{coin}macd-period_{t}"] = macd["macd"]
+
+                informative[f"%-{coin}relative_volume-period_{t}"] = (
+                    informative["volume"] / informative["volume"].rolling(t).mean()
+                )
+
+            informative[f"%-{coin}pct-change"] = informative["close"].pct_change()
+            informative[f"%-{coin}raw_volume"] = informative["volume"]
+            informative[f"%-{coin}raw_price"] = informative["close"]
+
+            indicators = [col for col in informative if col.startswith("%")]
+            # This loop duplicates and shifts all indicators to add a sense of recency to data
+            for n in range(self.freqai_info["feature_parameters"]["shift"] + 1):
+                if n == 0:
+                    continue
+                informative_shift = informative[indicators].shift(n)
+                informative_shift = informative_shift.add_suffix("_shift-" + str(n))
+                informative = pd.concat((informative, informative_shift), axis=1)
+
+            df = merge_informative_pair(df, informative, self.config["timeframe"], tf, ffill=True)
+            skip_columns = [
+                (s + "_" + tf) for s in ["date", "open", "high", "low", "close", "volume"]
+            ]
+            df = df.drop(columns=skip_columns)
+
+            # Add generalized indicators here (because in live, it will call this
+            # function to populate indicators during training). Notice how we ensure not to
+            # add them multiple times
+            if set_generalized_indicators:
+                df["%-day_of_week"] = (df["date"].dt.dayofweek + 1) / 7
+                df["%-hour_of_day"] = (df["date"].dt.hour + 1) / 25
+
+                # user adds targets here by prepending them with &- (see convention below)
+                # If user wishes to use multiple targets, a multioutput prediction model
+                # needs to be used such as templates/CatboostPredictionMultiModel.py
+                df["&-s_close"] = (
+                    df["close"]
+                    .shift(-self.freqai_info["feature_parameters"]["period"])
```

Comment:
shouldn't `shift()` here be using the "shift" parameter? that would seem to make more sense to me at first glance ... ?

Reply:
i did understand that - it solely wasn't clear why we'd shift it back by the same amount than "rolling" - it's however clear now after thinking about it a bit more